core/core.js

```diff
@@ -0,0 +1,164 @@
+'use strict'; // For testing with node.js
+
+// Global scope declarations:
+var $ = Object.create(Object.prototype);
+
+(function() {
+  // Core build options:
+
+  // Overwrite existing?:
+  var force = false;
+
+  /**
+   * Add a property to a core object - but only overwrite existing
+   * properties if force is true.
+   * @param {Object} obj The object to which to add a property.
+   * @param {string} key The property key to add.
+   * @param {*} value The property key to add.
+   */
+  function make(obj, key, value) {
+    if (force || typeof obj[key] === 'undefined') {
+      obj[key] = value;
+    }
+  }
+
+  /*******************************************************************
+   * $.object
+   */
+  make($, 'object', Object.prototype);
+
+  /*******************************************************************
+   * $.physical
+   */
+  make($, 'physical', Object.create($.object, {
+    name: {value: 'Physical object prototype',
+	   writable: true, enumerable: true, configurable: true},
+    location: {value: null,
+	       writable: true, enumerable: true, configurable: true},
+    contents_: {value: undefined, // wil be vetted into existance
+		writable: true, enumerable: true, configurable: true},
+  }));
+
+  make($.physical, 'contents', function() {
+    // Return the contents of this object.  This is for VR purposes,
+    // and descendents of $.physical can override it to 'hide' certain
+    // objects from their contents.
+    $.physical_vet(this);
+    return this.contents_;
+  });
+       
+  make($.physical, 'moveto', function(dest) {
+    // Move this physical object to dest
+    //
+    // This is based loosely on moo.ca's #3:moveto and #102:bf_move
+    // Vet obj, src and dest (also does type check):
+    $.physical_vet(this);
+    var src = this.location;
+    src === null || $.physical_vet(src);
+    dest === null || $.physical_vet(dest);
+
+    // FIXME: permission checks go here.
+    // Do moveable and accept checks:
+    if (!this.moveable(dest)) {
+      var destname = (dest === null ? 'null' : dest.name);
+      throw Error(this.name + ' declined to move to ' + destname);
+    }
+    if (dest !== null && !dest.accept(this)) {
```

Comment:
How about just !dest, instead of a null-only check.  Or: !dest || !dest.accept || ...